App.js

```diff
@@ -132,17 +133,19 @@ const App = () => {
       <RecoilRoot>
         {__DEV__ ? <RecoilDebugObserver instance={reactotronInstance} /> : null}
         <Store>
-          <NavigationContainer
-            onReady={onReadyState}
-            onStateChange={handleStateChange}
-            ref={navigationRef}
-            linking={linking}>
-            <View>
-              <OverlayProvider topInset={top} bottomInset={bottom} i18nInstance={streami18n}>
-                <RootNavigator areaHeight={height} currentScreen={currentScreen} />
-              </OverlayProvider>
-            </View>
-          </NavigationContainer>
+          <GestureHandlerRootView style={{flex: 1}}>
+            <NavigationContainer
+              onReady={onReadyState}
+              onStateChange={handleStateChange}
+              ref={navigationRef}
+              linking={linking}>
+              <View>
+                <OverlayProvider topInset={top} bottomInset={bottom} i18nInstance={streami18n}>
+                  <RootNavigator areaHeight={height} currentScreen={currentScreen} />
+                </OverlayProvider>
+              </View>
+            </NavigationContainer>
+          </GestureHandlerRootView>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The introduction of `GestureHandlerRootView` from `react-native-gesture-handler` is a good addition for handling complex gestures in the application. However, please ensure that all child components within this view are compatible with it and can properly propagate touch events. Also, make sure to test thoroughly across different devices and screen sizes to ensure consistent behavior.
```diff
-          <NavigationContainer
-            onReady={onReadyState}
-            onStateChange={handleStateChange}
-            ref={navigationRef}
-            linking={linking}>
-            <View>
-              <OverlayProvider topInset={top} bottomInset={bottom} i18nInstance={streami18n}>
-                <RootNavigator areaHeight={height} currentScreen={currentScreen} />
-              </OverlayProvider>
-            </View>
-          </NavigationContainer>
+          <GestureHandlerRootView style={{flex: 1}}>
+            <NavigationContainer
+              onReady={onReadyState}
+              onStateChange={handleStateChange}
+              ref={navigationRef}
+              linking={linking}>
+              <View>
+                <OverlayProvider topInset={top} bottomInset={bottom} i18nInstance={streami18n}>
+                  <RootNavigator areaHeight={height} currentScreen={currentScreen} />
+                </OverlayProvider>
+              </View>
+            </NavigationContainer>
+          </GestureHandlerRootView>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->